app/models/registration.rb

```diff
@@ -445,4 +455,44 @@ def ensure_waitlist_eligibility!
   def serializable_hash(options = nil)
     super(DEFAULT_SERIALIZE_OPTIONS.merge(options || {}))
   end
+
+  def auto_accept
+    return log_error('Auto-accept is not enabled for this competition.') unless competition.auto_accept_registrations
+    return log_error('Can only auto-accept pending registrations or first position on waiting list') unless
+      competing_status_pending? || (competing_status_waiting_list? && waiting_list_position == 1)
+    return log_error("Competition has reached auto_accept_disable_threshold of #{competition.auto_accept_disable_threshold} registrations") if
+      competition.auto_accept_threshold_reached?
+    return log_error('Competitor still has outstanding registration fees') if outstanding_entry_fees > 0
+    return log_error('Cant auto-accept while registration is not open') if !competition.registration_currently_open?
+
+    if competition.registration_full_and_accepted? && competing_status_pending?
+      update_lanes!(
+        { user_id: user_id, competing: { status: Registrations::Helper::STATUS_WAITING_LIST } }.with_indifferent_access,
+        'Auto-accept',
+      )
+    else
+      update_lanes!(
+        { user_id: user_id, competing: { status: Registrations::Helper::STATUS_ACCEPTED } }.with_indifferent_access,
+        'Auto-accept',
+      )
+    end
+  rescue ActiveRecord::RecordInvalid => e
+    log_error("Auto accept error for registration #{id}: #{e}")
```

Comment:
Nope. We don't catch "raw" errors like that. Instead, we fix what's causing the error :upside_down_face: 
(as an alternative, soft `update` instead of `update!` returns a boolean that indicates whether the save operation was successful or not)

Reply:
The problem is we don't necessarily know what might cause the error, and if not for catching and logging the error, we would have no visibility of that error at all. 

The call chain is: 
- update_lanes!
- Registrations::Lanes::Competing.update! [note that `update!` is NOT an ActiveRecord method - it is a custom-defined function in competing.rb - maybe it needs to be renamed to `update_lane`?]
- registration.save!

We could instead call `registration.save`, and then perhaps propagate the `registration` object up the call chain so that we can call `registration.errors.full_messages` on it to populate the log_error call - but that might be downstream of the log_error discussion